src/validator/rpc/interface.ts

```diff
@@ -0,0 +1,62 @@
+import {BeaconBlock, BeaconState, BLSPubkey, Epoch, Shard, Slot, ValidatorIndex} from "../../types";
+import {CommitteeAssignment, GenesisInfo} from "../types";
+
+export interface RpcClient {
```

Comment:
I think this interface should not redefine the API of BeaconAPI, but rather inherit from it. If the validator needs more functionality, we should extend BeaconAPI (or ValidatorAPI or whatever we decide to call it).
The only thing we should need to extend/switch on here is how the RPC is communicated.

Reply:
Agreed. I think validator now only needs ValidatorApi so Rpc client should only extend ValidatorApi interface.